api-gateway-routes/api_gateway_proxy_function.rb

```diff
@@ -62,7 +64,24 @@ def on_connect(event, context)
 
 def on_disconnect(event, context)
   sqs = Aws::SQS::Client.new(region: get_region(context))
-  sqs.delete_queue(queue_url: get_sqs_url(event, context))
+
+  # Handle if queue does not exist,
+  # such as in case of connectivity test.
+  # The NonExistentQueue error is not documented in the AWS SDK,
+  # but was observed in errors accumulated from our connectivity tests.
+  begin
+    sqs.delete_queue(queue_url: get_sqs_url(event, context))
+  rescue Aws::SQS::Errors::NonExistentQueue => e
```

Comment:
I found a project rescuing this (undocumented) error as far back as 2015, so I'm thinking it's pretty stable:

https://github.com/ruby-shoryuken/shoryuken/issues/150